src/NServiceBus.AcceptanceTests/Recoverability/Retries/When_message_is_moved_to_error_queue.cs

```diff
@@ -0,0 +1,168 @@
+﻿namespace NServiceBus.AcceptanceTests.Recoverability.Retries
+{
+    using System;
+    using System.Threading.Tasks;
+    using AcceptanceTesting;
+    using EndpointTemplates;
+    using Features;
+    using NServiceBus.Pipeline;
+    using NUnit.Framework;
+    using ScenarioDescriptors;
+
+    public class When_message_is_moved_to_error_queue : NServiceBusAcceptanceTest
+    {
+        [TestCase(TransportTransactionMode.ReceiveOnly)]
+        [TestCase(TransportTransactionMode.SendsAtomicWithReceive)]
+        [TestCase(TransportTransactionMode.TransactionScope)]
+        public async Task Should_not_send_outgoing_messages(TransportTransactionMode transactionMode)
+        {
+            await Scenario.Define<Context>(c =>
+            {
+                c.Id = Guid.NewGuid();
+                c.TransactionMode = transactionMode;
+            })
+            .WithEndpoint<Endpoint>(b => b.DoNotFailOnErrorMessages()
+                .When((session, context) => session.SendLocal(new InitiatingMessage
+                {
+                    Id = context.Id
+                }))
+            )
+            .WithEndpoint<ErrorSpy>()
+            .Done(c => c.MessageMovedToErrorQueue)
+            .Repeat(r => r.For<AllDtcTransports>())
+            .Should(c => Assert.IsFalse(c.OutgoingMessageSent, "Outgoing messages should not be sent"))
+            .Run();
+        }
+
+        [Test]
+        public async Task May_send_outgoing_messages_without_transport_transactions()
```

Comment:
What would the benefit be? This test is essentialy testing the `TransportTransactionMode.None` case missing from the test above.